flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/BootstrapTools.java

```diff
@@ -432,7 +432,12 @@ public static String getTaskManagerShellCommand(
 				tmParams.taskManagerDirectMemoryLimitMB()));
 		}
 
-		startCommandValues.put("jvmmem", StringUtils.join(params, ' '));
+		final TaskExecutorResourceSpec taskExecutorResourceSpec = tmParams.getTaskExecutorResourceSpec();
+		if (taskExecutorResourceSpec == null) { // FLIP-49 disabled
```

Comment:
`taskExecutorResourceSpec` should never be `null` after enabling FLIP-49. The only reason it might be `null` is that we passed in `null` value for the legacy code paths. So I think it should not be decoupled.